packages/stores/src/ui-config/manage-liquidity/add-concentrated-liquidity.ts

```diff
@@ -0,0 +1,313 @@
+import { TxChainSetter } from "@keplr-wallet/hooks";
+import {
+  ChainGetter,
+  IQueriesStore,
+  ObservableQueryBalances,
+} from "@keplr-wallet/stores";
+import { Dec, Int } from "@keplr-wallet/unit";
+import {
+  ActiveLiquidityPerTickRange,
+  maxSpotPrice,
+  minSpotPrice,
+  priceToTick,
+  roundPriceToNearestTick,
+} from "@osmosis-labs/math";
+import { ConcentratedLiquidityPool } from "@osmosis-labs/pools";
+import { action, makeObservable, observable } from "mobx";
+
+import {
+  PriceRange,
+  TokenPairHistoricalPrice,
+} from "../../queries-external/token-pair-historical-chart/types";
+
+/** Use to config user input UI for eventually sending a valid add concentrated liquidity msg.
+ */
+export class ObservableAddConcentratedLiquidityConfig extends TxChainSetter {
+  @observable
+  protected _poolId: string;
+
+  @observable
+  protected _sender: string;
+
+  @observable
+  protected _pool: ConcentratedLiquidityPool;
+
+  /*
+	 Used to get current view type of AddConcLiquidity modal
+	 */
+  @observable
+  protected _modalView: "overview" | "add_manual" | "add_managed" = "overview";
+
+  /*
+   Used to get historical range for price chart
+   */
+  @observable
+  protected _historicalRange: PriceRange = "7d";
+
+  /*
+    Used to get historical data for price chart
+  */
+  @observable
+  protected _historicalChartData: TokenPairHistoricalPrice[] = [];
+
+  /*
+    Used to get active liquidity data
+  */
+  @observable
+  protected _activeLiquidity: ActiveLiquidityPerTickRange[] = [];
+
+  /*
+   Used to get min and max range for adding concentrated liquidity
+   */
+  @observable
+  protected _range: [Dec, Dec] = [minSpotPrice, minSpotPrice];
+
+  @observable
+  protected _fullRange: boolean = false;
+
+  @observable
+  protected _zoom: number = 1;
+  /*
+   Used to get base and quote asset deposit for adding concentrated liquidity
+   */
+  @observable
+  protected _baseDepositAmountIn: Dec = new Dec(0);
+
+  @observable
+  protected _quoteDepositAmountIn: Dec = new Dec(0);
+
+  constructor(
+    chainGetter: ChainGetter,
+    initialChainId: string,
+    poolId: string,
```

Comment:
Everything that is expected to remain the same for the life of this UI config, like the pool, should not be observable with getters/setters. (This applies to the members above). For this poolId specifically, we could remove a lot of code by using `protected readonly` to overload it as a member.